azure_sdk/lib/Profile_Azure_Stack/modules/compute_profile_module.rb

```diff
@@ -0,0 +1,666 @@
+      OperatingSystemTypes = Azure::ARM::Compute::Api_2017_03_30::Models::OperatingSystemTypes
+      VirtualMachineSizeTypes = Azure::ARM::Compute::Api_2017_03_30::Models::VirtualMachineSizeTypes
+      CachingTypes = Azure::ARM::Compute::Api_2017_03_30::Models::CachingTypes
+      DiskCreateOptionTypes = Azure::ARM::Compute::Api_2017_03_30::Models::DiskCreateOptionTypes
+      StorageAccountTypes = Azure::ARM::Compute::Api_2017_03_30::Models::StorageAccountTypes
+      PassNames = Azure::ARM::Compute::Api_2017_03_30::Models::PassNames
+      ComponentNames = Azure::ARM::Compute::Api_2017_03_30::Models::ComponentNames
+      SettingNames = Azure::ARM::Compute::Api_2017_03_30::Models::SettingNames
+      ProtocolTypes = Azure::ARM::Compute::Api_2017_03_30::Models::ProtocolTypes
+      ResourceIdentityType = Azure::ARM::Compute::Api_2017_03_30::Models::ResourceIdentityType
+      MaintenanceOperationResultCodeTypes = Azure::ARM::Compute::Api_2017_03_30::Models::MaintenanceOperationResultCodeTypes
+      UpgradeMode = Azure::ARM::Compute::Api_2017_03_30::Models::UpgradeMode
+      OperatingSystemStateTypes = Azure::ARM::Compute::Api_2017_03_30::Models::OperatingSystemStateTypes
+      ResourceSkuCapacityScaleType = Azure::ARM::Compute::Api_2017_03_30::Models::ResourceSkuCapacityScaleType
+      ResourceSkuRestrictionsType = Azure::ARM::Compute::Api_2017_03_30::Models::ResourceSkuRestrictionsType
+      ResourceSkuRestrictionsReasonCode = Azure::ARM::Compute::Api_2017_03_30::Models::ResourceSkuRestrictionsReasonCode
+      IPVersion = Azure::ARM::Compute::Api_2017_03_30::Models::IPVersion
+      VirtualMachineScaleSetSkuScaleType = Azure::ARM::Compute::Api_2017_03_30::Models::VirtualMachineScaleSetSkuScaleType
+      DiskCreateOption = Azure::ARM::Compute::Api_2017_03_30::Models::DiskCreateOption
+      AccessLevel = Azure::ARM::Compute::Api_2017_03_30::Models::AccessLevel
+      InstanceViewTypes = Azure::ARM::Compute::Api_2017_03_30::Models::InstanceViewTypes
+    end
+
+    #
+    # Compute
+    #
+    class ComputeClass
+      attr_accessor :availability_sets, :virtual_machine_extension_images, :virtual_machine_extensions, :virtual_machine_images, :usage_operations, :virtual_machine_sizes, :images, :resource_skus, :virtual_machines, :virtual_machine_scale_sets, :virtual_machine_scale_set_extensions, :virtual_machine_scale_set_vms, :disks, :snapshots, :virtual_machine_run_commands, :configurable, :base_url, :options, :model_classes
+
+      def initialize(configurable, base_url, options)
+        @configurable, @base_url, @options = configurable, base_url, options
+        client = Azure::ARM::Compute::Api_2017_03_30::ComputeManagementClient.new(configurable.credentials, base_url, options)
+        if(client.respond_to?(:subscription_id))
+          client.subscription_id = configurable.subscription_id
+        end
+        self.availability_sets = Azure::ARM::Compute::Api_2017_03_30::AvailabilitySets.new(client)
```

Comment:
Because, when you want to use multiple versions, you will have multiple clients defined and then use those clients to create the operation objects. That is the reason I am following this approach.